nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -340,8 +340,23 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
 #endif
 
   // Smooth plan
-  if (num_iterations > 1 && plan.poses.size() > 6) {
-    _smoother->smooth(plan, costmap, time_remaining);
+  // Check if there are any cusps and if so, split path to avoid smoothing cusps
+  if (num_iterations > 1) {
+    std::vector<unsigned int> cusps_id = findDirectionChange(plan);
+    if (cusps_id.size() < 3 && plan.poses.size() > 6) {
```

Comment:
Why `cusps_id.size() < 3`?

Reply:
The vector cusps_id always contains at least the indices of the initial and final poses, thus if no cusps changes of direction were found the vector size would be 2. I added some comments in the code regarding this

Reply:
Why return the first and last points? Why would those be considered cusps?